bothan-api/server-cli/src/commands/query.rs

```diff
@@ -0,0 +1,424 @@
+async fn query_bybit<T: Into<Duration>>(
+    opts: bothan_bybit::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let connector = bothan_bybit::api::WebSocketConnector::new(opts.url);
+    let asset_infos = query_single_websocket(connector, query_ids, timeout.into()).await?;
+
+    display_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_coinbase<T: Into<Duration>>(
+    opts: bothan_coinbase::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let connector = bothan_coinbase::WebSocketConnector::new(opts.url);
+    let asset_infos = query_websocket_in_chunks(
+        connector,
+        query_ids,
+        opts.max_subscription_per_connection,
+        timeout.into(),
+    )
+    .await?;
+
+    display_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_coingecko<T: Into<Duration>>(
+    opts: bothan_coingecko::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let api = bothan_coingecko::api::RestApiBuilder::new(opts.url, opts.user_agent, opts.api_key)
+        .build()?;
+    let asset_infos = query_rest_api(&api, query_ids, timeout.into()).await?;
+
+    display_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_coinmarketcap<T: Into<Duration>>(
+    opts: bothan_coinmarketcap::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let api = bothan_coinmarketcap::api::RestApiBuilder::new(opts.url, opts.api_key).build()?;
+    let asset_infos = query_rest_api(&api, query_ids, timeout.into()).await?;
+
+    display_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_htx<T: Into<Duration>>(
+    opts: bothan_htx::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let connector = bothan_htx::api::WebSocketConnector::new(opts.url);
+    let asset_infos = query_single_websocket(connector, query_ids, timeout.into()).await?;
+
+    display_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_kraken<T: Into<Duration>>(
+    opts: bothan_kraken::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let connector = bothan_kraken::api::WebSocketConnector::new(opts.url);
+    let asset_infos = query_single_websocket(connector, query_ids, timeout.into()).await?;
+
+    display_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_okx<T: Into<Duration>>(
+    opts: bothan_okx::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let connector = bothan_okx::api::WebSocketConnector::new(opts.url);
+    let asset_infos = query_single_websocket(connector, query_ids, timeout.into()).await?;
+
+    display_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_rest_api<P: RestAssetInfoProvider>(
+    provider: &P,
+    query_ids: &[String],
+    timeout_interval: Duration,
+) -> anyhow::Result<Vec<AssetInfo>>
+where
+    P::Error: StdError + Send + Sync + 'static,
+{
```

Comment:
please dont combine generic parameters in the angle brackets when using where, choose one or the other